homeassistant/components/stiebel_eltron/climate.py

```diff
@@ -0,0 +1,153 @@
+"""Support for stiebel_eltron climat platform."""
+import logging
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_HUMIDITY, STATE_AUTO, STATE_IDLE, STATE_MANUAL,
+    SUPPORT_OPERATION_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.components.stiebel_eltron import DOMAIN as STE_DOMAIN
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+DEPENDENCIES = ['stiebel_eltron']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
+
+STATE_DAYMODE = 'day_mode'
+STATE_SETBACK = 'setback_mode'
+STATE_DHW = 'dhw'
+STATE_EMERGENCY = 'emergency'
```

Comment:
Only the states present in the base climate component are allowed as operation mode states. Please remove these other states here.

Reply:
Why are no specific states allowed? This is not just a normal air conditioning device. It's a heat pump with integrated domestic hot water (dhw) module.

Thus, I either remove all specific states or I try to map them to the provided states in the base component (like for [KNX Climate](https://www.home-assistant.io/components/climate.knx/) or [Honeywell evohome/TCC systems](https://www.home-assistant.io/components/evohome/). What do you recommend?

Reply:
It's ok to map the device modes to the existing home assistant climate states however possible. We just can't introduce more home assistant states.

We have to keep the device interface that home assistant provides to the user consistent. This allows to integrate devices with automations, frontend and cloud services.

Reply:
Consider implementing the dhw part as a `water_heater` platform.

Reply:
That's a good idea. I did not know the `water_heater` platform. I will do that in another PR. I have removed the custom states now and introduced a mapping.